int16[] color_code
---
int32[] color_occurrence
